kartothek/core/dataset.py

```diff
@@ -376,29 +376,36 @@ def get_indices_as_dataframe(
         Parameters
         ----------
         """
+        if not self.primary_indices_loaded:
+            # self.load_partition_indices is not inplace
```

Comment:
I guess this is why we implemented this differently so far. That's fine for me, though.